5 4 add screen, search screen

Sotonka opened the pull request:
![Screenshot_1670191069](https://user-images.githubusercontent.com/71734734/205517998-aeea40c7-da88-46ca-8c0e-5f5d5f64e92c.png)

![Screenshot_1670191073](https://user-images.githubusercontent.com/71734734/205517924-5f8ba14c-2363-49bb-8cdd-e36143629974.png)

![Screenshot_1670191079](https://user-images.githubusercontent.com/71734734/205517931-dac831ba-0aa2-4146-b784-f894cf630bd3.png)

![Screenshot_1670191145](https://user-images.githubusercontent.com/71734734/205517939-4042e83c-8ac5-4053-aa28-bda1fe2faf02.png)
![Screenshot_1670191153](https://user-images.githubusercontent.com/71734734/205517941-23217dee-06c6-444e-bd73-003003a0fd4f.png)

![Screenshot_1670191181](https://user-images.githubusercontent.com/71734734/205517946-6dbc130f-8490-4f51-9cef-404bd3179bc4.png)

![Screenshot_1670191208](https://user-images.githubusercontent.com/71734734/205518265-fdc43066-3d26-4a9a-bde2-1f9af67986d0.png)

Sotonka committed:
переделал ui kit

Sotonka committed:
добавление Sight

Sotonka committed:
передача фильтров на главный экран

Sotonka committed:
передаю фильтр вместо List<Sight>

Sotonka committed:
экран поиска

internetova commented on lib/ui/screen/filters_screen.dart:
```diff
@@ -108,16 +99,14 @@ class _BuildAppBar extends StatelessWidget implements PreferredSizeWidget {
           children: [
             InkWell(
```
Можно сделать отдельный компонент-виджет, он же у вас не в одном месте будет. Можно добавить его и на страницу с поиском.

Sotonka replied:
пока не знаю до какого момента еще подождать - вообще фильтры по другому хочу сделать, и экраны с поиском и со списком объединить

internetova commented on lib/ui/screen/filters_screen.dart:
```diff
@@ -145,27 +134,27 @@ class _BuildCategories extends StatelessWidget {
     final categoryRow1 = <_Category>[
       _Category(
         type: SightType.hotel,
         name: 'Отель',
```
хорошо бы вынести все тексты в отдельный класс, AppStrings

Sotonka replied:
пока оставил так, проще ориентироваться

internetova commented on lib/ui/screen/filters_screen.dart:
```diff
@@ -145,27 +134,27 @@ class _BuildCategories extends StatelessWidget {
-    final themeColors = Theme.of(context).extension<ThemeColors>()!;
+    final themeColors = Theme.of(context).extension<AppThemeColors>()!;
     final filter = context.watch<FilterProvider>().filter;
     final categoryRow1 = <_Category>[
```
Лучше:
1. Сделать отдельный класс для категорий
2. Для отображения использовать пока виджет Wrap, а не 2 row
3. Выводить не руками каждый пункт, а использовать map

Основные методы работы с коллекциями
https://youtu.be/W-gIKcS_hQA

Sotonka replied:
V

internetova commented on lib/ui/screen/filters_screen.dart:
```diff
@@ -263,20 +252,20 @@ class _BuildSlider extends StatelessWidget {
                 style: theme.primaryTextTheme.headline6,
               ),
               Text(
                 'от ${(filter.distance.start / 1000).round()} до ${(filter.distance.end / 1000).round()} км',
```
это `(filter.distance.start / 1000).round()` лучше под фильтром сделать:
```dart
final distanceStart = filter.distance.start / 1000).round();

Sotonka replied:
V

internetova commented on lib/ui/screen/settings_screen.dart:
```diff
@@ -17,20 +16,24 @@ class _SettingsScreenState extends State<SettingsScreen> {
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
-    final themeColors = Theme.of(context).extension<ThemeColors>()!;
+    final themeColors = Theme.of(context).extension<AppThemeColors>()!;
```
вы же уже получили тему, можно писать так:
```dart
final themeColors = theme.extension<AppThemeColors>()!;

Sotonka replied:
V

internetova commented on lib/ui/screen/sight_details.dart:
```diff
@@ -14,7 +14,7 @@ class SightDetails extends StatelessWidget {
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
-    final themeColors = theme.extension<ThemeColors>()!;
+    final themeColors = theme.extension<AppThemeColors>()!;
```
👆

Sotonka replied:
V

internetova commented on lib/ui/screen/sight_list_screen.dart:
```diff
@@ -1,42 +1,181 @@
   }
 }
+
+class SightListProvider with ChangeNotifier {
```
1. Лучше сделать папку data_providers и складывать туда
2. with -> extends

Sotonka replied:
V

internetova commented on lib/ui/screen/sight_search_screen.dart:
```diff
@@ -0,0 +1,356 @@
+    return Consumer<SearchProvider>(builder: (context, provider, child) {
+      return provider.searchResult == null
+          // ЗАМЕНИТЬ НА ЭКРАН НИЧЕГО НЕ НАЙДЕНО
+          ? const SizedBox()
```
вот да))

Sotonka replied:
V

internetova commented on lib/ui/widget/gradient_button.dart:
```diff
@@ -0,0 +1,60 @@
+    return InkWell(
+      onTap: onPressed,
+      child: Container(
+        //constraints: BoxConstraints(minHeight: 10, maxWidth: 10),
```
добавляйте todo или удаляйте

Sotonka replied:
V

internetova commented on lib/ui/widget/nothing_found.dart:
```diff
@@ -0,0 +1,42 @@
+          ),
+          const SizedBox(height: 32),
+          Text(
+            'Ничего не найдено.',
```
👆

Sotonka replied:
V

internetova commented on lib/ui/widget/nothing_found.dart:
```diff
@@ -0,0 +1,42 @@
+          ),
+          const SizedBox(height: 8),
+          Text(
+            'Попробуйте изменить параметры поиска',
```
👆

Sotonka replied:
V

internetova commented on lib/ui/widget/search_bar.dart:
```diff
@@ -0,0 +1,85 @@
+  final TextEditingController? controller;
+  final FocusNode? focus;
+  final VoidCallback? onComplete;
+  final void Function(String)? onChange;
```
давайте заменим на `ValueChanged<String>?`

Sotonka replied:
V

internetova commented on lib/ui/widget/search_bar.dart:
```diff
@@ -0,0 +1,85 @@
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final themeColors = Theme.of(context).extension<AppThemeColors>()!;
```
👆

Sotonka replied:
V

internetova commented on lib/ui/widget/search_bar.dart:
```diff
@@ -0,0 +1,85 @@
+        onTap: onPressed,
+        onEditingComplete: onComplete,
+        decoration: InputDecoration(
+          hintText: 'Поиск',
```
👆

Sotonka replied:
V

internetova commented on lib/ui/widget/sight_card_tab.dart:
```diff
@@ -0,0 +1,105 @@
+class SightCardTab extends StatelessWidget {
+  final Sight sight;
+  final String? search;
+  const SightCardTab({
```
почему не делаем отступы?

Sotonka replied:
V

internetova commented on lib/ui/widget/sight_card_tab.dart:
```diff
@@ -0,0 +1,105 @@
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final themeColors = Theme.of(context).extension<AppThemeColors>()!;
```
👆

Sotonka replied:
V

internetova commented on lib/ui/widget/sight_card_tab.dart:
```diff
@@ -0,0 +1,105 @@
+import 'package:places/ui/widget/loadable_image.dart';
+import 'package:places/utils/utils.dart';
+
+class SightCardTab extends StatelessWidget {
```
хорошо бы добавлять доки к классам и методам

Sotonka replied:
подскажите как правильно их оформлять

internetova replied:
@Sotonka открываете любой класс во фреймворке и смотрите 🤗
Чтобы проще было ориентироваться в коде другим разработчикам да и самому не забыть через какое-то время что вы понаписали хорошо давать краткое описание класса, а также методам и неочевидным свойствам

internetova reviewed (commented):
Надо доработать 🤗

**Экран Избранное**
Некорректно работаю табы

![2022-12-07 01 23 40](https://user-images.githubusercontent.com/39735343/206036779-59a52a84-f8b4-411c-ab7c-9792d7c26839.gif)

Таббар пример реализации:
https://medium.com/codechai/flutter-boring-tab-to-cool-tab-bfcb1a93f8d0

**Экран Фильтр**

Когда результатов 0 кнопка должна быть неактивной - надо поправить

**Экран поиск**

1. Надо доработать логику - ищет только по началу, а если в следующем слове не находит

2. Если ничего не найдено надо отображать это состояние - смотрим в дизайн

3. Хорошо бы добавить возможность возврата на предыдущий экран тапом по заголовку или иконку назад сделать (в дизайне упустили)


**Экран Добавить место**

1. Описание если не заполнено тоже надо подсветить красной рамкой

2. Не сохраняется выбранная категория при повторном возврате к выбору категорий
3. Нет кнопок очистить в полях Название и Описание
4. При вводе текста состояние поля с ошибочного на правильное не меняется - как у вас пользователь поймет что он ввел правильные данные?
5. Разного цвета кнопки очистки полей координат
6. При выбранной категории пишет Выберите категорию и не дает создать место
7. После создания надо показывать Алерт что место добавлено и уходить на главную страницу (там должно быть новое место - проверить не удалось)

![2022-12-07 00 37 54](https://user-images.githubusercontent.com/39735343/206037201-a395b6a5-651f-4e00-999a-6fe94da00142.gif)

<img width="315" alt="Снимок экрана 2022-12-07 в 00 36 01" src="https://user-images.githubusercontent.com/39735343/206037304-48c5da07-b2c8-41a2-b293-3bc4ecdcde45.png">
<img width="315" alt="Снимок экрана 2022-12-07 в 00 40 34" src="https://user-images.githubusercontent.com/39735343/206037309-54a36c95-305f-42d9-a647-632496358d5b.png">
<img width="359" alt="Снимок экрана 2022-12-07 в 00 41 37" src="https://user-images.githubusercontent.com/39735343/206037313-8f7234d5-70d6-49d7-8976-3efde3709aef.png">
<img width="315" alt="Снимок экрана 2022-12-07 в 00 43 19" src="https://user-images.githubusercontent.com/39735343/206037316-b987e4f6-2fc6-482a-8134-91b04d8704c3.png">


**Анализатор кода**
Должен быть пустой
<img width="1330" alt="Снимок экрана 2022-12-07 в 01 06 27" src="https://user-images.githubusercontent.com/39735343/206035626-4a833265-a22a-445b-a0d2-03b836462dd6.png">

А также поправить по комментариям

Sotonka commented:
**Экран Избранное**

1.Некорректно работаю табы
**_исправил_**

**Экран Фильтр**

1. Когда результатов 0 кнопка должна быть неактивной - надо поправить
**_исправил_**

**Экран поиск**

1. Надо доработать логику - ищет только по началу, а если в следующем слове не находит
**_так и было задумано)
переделал, используя либу_** 
https://pub.dev/packages/text_chunk_styling
**_если есть что-то получше/ другие способы это реализовать - интересно было бы узнать_**

3. Если ничего не найдено надо отображать это состояние - смотрим в дизайн
**_исправил_**

4. Хорошо бы добавить возможность возврата на предыдущий экран тапом по заголовку или иконку назад сделать (в дизайне упустили)
**_исправил_**

**Экран Добавить место**

1. Описание если не заполнено тоже надо подсветить красной рамкой
**_исправил_**

2. Не сохраняется выбранная категория при повторном возврате к выбору категорий
**_исправил_**

3. Нет кнопок очистить в полях Название и Описание
**_исправил_**

4. При вводе текста состояние поля с ошибочного на правильное не меняется - как у вас пользователь поймет что он ввел правильные данные?
**_исправил_**

5. Разного цвета кнопки очистки полей координат
**_исправил_**

6. При выбранной категории пишет Выберите категорию и не дает создать место
**_исправил_**

7. После создания надо показывать Алерт что место добавлено и уходить на главную страницу (там должно быть новое место - проверить не удалось)
**_исправил_**

Sotonka committed:
правки по заданию #1

-рабочая-
исправлены фильтры,
взаимодействие sightlistscreen - filters - searchscreen

Sotonka committed:
правки по заданию - экран поиска

Sotonka committed:
правки по заданию - добавление места

internetova commented on lib/domain/sight.dart:
```diff
@@ -40,4 +40,6 @@ final Map<String, String> _typeTranslations = {
   'SightType.restaurant': 'ресторан',
```
Все тексты лучше выносить в отдельный файл в AppStrings

internetova commented on lib/ui/providers/add_sight_provider.dart:
```diff
@@ -0,0 +1,235 @@
+
+  String? validateCategory(String? value) {
+    if (value == null || value.isEmpty) {
+      return 'Выберите категорию';
```
👆

internetova reviewed (commented):
Надо немного доработать:

1. Вынести все тексты в AppStrings
2. Поправить логику работы на фильтре. При клике на Очистить на кнопке есть результат 9 мест, хотя на фильтре ничего не выбрано

![2022-12-14 23 41 16](https://user-images.githubusercontent.com/39735343/207710856-9bebbee9-08a2-4afe-9ac1-9b506121723e.gif)

Sotonka committed:
правки по заданию

строки, комментарии

internetova reviewed (commented):
Вы вынесли не все тексты из кода. Например, под моим комментарием вы поставили галочку, а изменений нет 🤔 Предлагаю пройти по всем файлам и поправить.

Sotonka committed:
правки - вынес текст ы AppStrings
